rpc/types/events.go

```diff
@@ -97,40 +83,38 @@ func ParseTxResult(result *abci.ResponseDeliverTx) (*ParsedTxs, error) {
 		TxHashes: make(map[common.Hash]int),
 	}
 	for _, event := range result.Events {
```

Comment:
Consider: extracting the format check, then seperate each formats into its own functional block, this will also make it easier to purge deprecated format eventually